### ISSUE TITLE
which problems have 2 matrices A, M for Ax = λ Mx, please ?

### ISSUE THREAD

**Issue by denis-bz** [ginkgo-project/ssget#6]
Can anyone here help me please: which problems have 2 matrices A, M for Ax = λ Mx
(the generalized eigenvalue problem) ?
I'd have thought that all "Structural problems" would -- not so.
Thanks,
cheers

**Comment by upsj** [comment]
I don't think we can get such information directly from the metadata. The only matrices labelled explicitly as Eigenvalue problems are the ones from the [Rommes](https://sparse.tamu.edu/Rommes) collection. Most of the other problems are either linear systems Ax=b, linear programming instances or graphs encoded as matrices.

**Comment by yhmtsai** [comment]
I do not how to produce the A, M from Rommes dataset. They give the formula and mtx, mtx_B, mtx_C, (and mtx_D, mtx_E optionally).
You can get this data list from
`./ssget -s '[[ "@kind" =~ "eigenvalue" ]]'`: Search the kind containing "eigenvalue"
or 
`./ssget -s '[[ "@group" == "Rommes" ]]'`: Search the Rommes group

**Comment by yhmtsai** [comment]
Unfortunately, we only give the main mtx path now.(`./ssget -i id -e`) now, so modifying the path to get other matrix (mtx_B, mtx_C...) is necessary.

**Comment by denis-bz** [comment]
Hi Ginkgo people,
  thanks for the info. It seems, though, that the Rommes problems are for
    E dx/dt = Ax + Bu; y = Cx + Du   not Ax = λ Mx.
(I'm afraid I know hardly anything about control theory).

For what it's worth, my standard sanity check
on Rommes/nopss_11k nopss_11k.mtx 11685 x 11685
shows a few very tiny and a few huge entries --

    np.percentile( abs( A.data ), q=[
    [0,   .1,   1,      99,  99.9, 100 ])
    9e-18 4e-09 0.012   2643 10123 1e+20

I'd expect such a huge range in values to break any linear solver :)
let alone eigensolvers, but I'm no expert.

If you know anyone who might know of problems for Ax = λ Mx, could you please ask them ?
Thanks again,
cheers

**Comment by yhmtsai** [comment]
Use bcsstk as A and related bcsstm as M for generalized eigenvalue problem.
Currently, I do not find the corresponding documentation.
Some related comments are in page 11 of this [paper](http://user.it.uu.se/~maya/Courses/NLA_Parallel/Slides_2013/Ghysels_12.2012.1.pdf)

I used this matrix pair before, but I totally forgot it.
